tests/tests_unit/test_cdf_tk/test_utils.py

```diff
@@ -384,14 +406,25 @@ def test_validate(
         expected_messages: list[str],
         expected_vars: dict[str, str],
     ) -> None:
-        with MonkeyPatch.context() as mp:
-            for key, value in environment_variables.items():
-                mp.setenv(key, value)
+
+        with mock.patch.dict(os.environ, environment_variables, clear=True):
             auth_var = AuthVariables.from_env()
             results = auth_var.validate(verbose)
 
             assert results.status == expected_status
             assert results.messages == expected_messages
 
-            if expected_vars:
-                assert vars(auth_var) == expected_vars
+            actual_vars = vars(auth_var)
+            for expected_var in expected_vars:
+                if expected_vars.get(expected_var) is not None:
+                    assert actual_vars.get(expected_var) == expected_vars.get(
+                        expected_var
+                    ), f"expected {expected_var} to be {expected_vars.get(expected_var)} but got {actual_vars.get(expected_var)}"
```

Comment:
I am not a big fan of for loops with asserts. If you check all at once you will be notified of all mismatching asserts at once, as this will force you as a developer to work in a loop: one falling -> you fix it -> next failing -> you fix it -> and so on.

Reply:
Agree, it was just for debugging. Reverted!